src/data_loading.py

```diff
@@ -332,36 +335,82 @@ def get_api_blood_data(api_root = 'https://api.a2cps.org/files/v2/download/publi
+    tapis_token = get_tapis_token(portal_api_root, coresessionid)
+    print(tapis_token)
+    auth_status = get_auth_status(vbr_api_root, tapis_token)
+
+    if auth_status == True:
+        try:
+            # Load Json Data
+            subjects1_filepath = '/'.join([api_root,'subjects','subjects-1-latest.json'])
+            subjects1_request = requests.get(subjects1_filepath)
+            if subjects1_request.status_code == 200:
+                subjects1 = subjects1_request.json()
+            else:
+                return None
+                # return {'status':'500', 'source': api_dict['subjects']['subjects1']}
 
-    try:
-        # Load Json Data
-        subjects1_filepath = '/'.join([api_root,'subjects','subjects-1-latest.json'])
-        subjects1_request = requests.get(subjects1_filepath)
-        if subjects1_request.status_code == 200:
-            subjects1 = subjects1_request.json()
-        else:
-            return None
-            # return {'status':'500', 'source': api_dict['subjects']['subjects1']}
-
-        subjects2_filepath = '/'.join([api_root,'subjects','subjects-2-latest.json'])
-        subjects2_request = requests.get(subjects2_filepath)
-        if subjects2_request.status_code == 200:
-            subjects2 = subjects2_request.json()
-        else:
-            return None
-            # return {'status':'500', 'source': api_dict['subjects']['subjects2']}
+            subjects2_filepath = '/'.join([api_root,'subjects','subjects-2-latest.json'])
+            subjects2_request = requests.get(subjects2_filepath)
+            if subjects2_request.status_code == 200:
+                subjects2 = subjects2_request.json()
+            else:
+                return None
+                # return {'status':'500', 'source': api_dict['subjects']['subjects2']}
 
-        # Create combined json
-        subjects_json = {'1': subjects1, '2': subjects2}
+            # Create combined json
+            subjects_json = {'1': subjects1, '2': subjects2}
 
-        return subjects_json
+            return subjects_json
 
-    except Exception as e:
-        traceback.print_exc()
+        except Exception as e:
+            traceback.print_exc()
+            return None
+    else:
+        print("Unauthorized attempt to access Subjects data")
         return None
-
+    
+def get_tapis_token(portal_api_root, coresessionid = None):
+    print('get_tapis_token')
+    print(coresessionid)
+    print('portal_api_root')
+    print(portal_api_root)
+    try:
+        response = requests.get(portal_api_root + '/auth/tapis/', headers={'cookie':'coresessionid=' + coresessionid})
+        print('portal api response:')
+        print(response)
+        if response:
+            tapis_token = response.json()['token']
+            return tapis_token
+        else:
+            print("Unauthorized to access tapis token")
+            raise Exception
+    except Exception as e:
+        return('portal api error: {}'.format(e))
```

Comment:
```suggestion
        print('portal api error: {}'.format(e))
        raise e
```
the calling of this method (~line 344) assumes that it returns a token. so best to raise an exception here.